components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -412,6 +412,8 @@ private OAuth2AccessTokenRespDTO generateNewAccessToken(OAuthTokenReqMessageCont
                 validityPeriodInMillis, oauthTokenIssuer);
         setDetailsToMessageContext(tokReqMsgCtx, validityPeriodInMillis, newTokenBean, timestamp);
 
+        setACRValueInTheToken(tokReqMsgCtx, newTokenBean);
```

Comment:
We might need to add the same change to the `generateAccessToken` function of the [ResponseTypeHandlerUtil](https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/c20168be014860b69fd642531e930f32b92ade55/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/util/ResponseTypeHandlerUtil.java#L123) class as well.